version: '3'

services:
  db:
    image: postgres:16.3-alpine
    environment:
      POSTGRES_HOST_AUTH_METHOD : trust
      PGDATA : /var/lib/postgresql/data/pgdata
    env_file:
        - .env
    ports:
      - "5433:5432"
    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U $POSTGRES_USER -d $POSTGRES_DB" ]
      interval: 10s
      timeout: 5s
      retries: 5

  redis:
    image: redis:5.0.4-alpine
    ports:
      - '6378:6379'

  app:
    build: .
    command: bash -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    ports:
      - '8000:8000'
    depends_on:
       - db
    volumes:
      - .:/app
    env_file:
      - .env


  celery:
    build: .
    tty: true
    command: celery -A config worker -l INFO
    depends_on:
      - redis
      - app
      - db
    env_file:
      - .env
    volumes:
      - .:/app

  celery_beat:
    build: .
    tty: true
    command: celery -A config beat -l INFO -S django
    depends_on:
      - redis
      - app
    env_file:
      - .env

volumes:
  pg_data: